File: CPP [3501-4000]/3550.smallest_index_with_digit_sum_equal_to_index [1].cpp
class Solution {
private:
    int digitSum(int num){
        int sum = 0;
        while(num > 0)
        {
            sum += num % 10;
            num /= 10;
        }

        return sum;
    }

public:
    int smallestIndex(vector<int>& nums) {
        for(int i=0; i<nums.size(); i++)
            if(digitSum(nums[i]) == i)
                return i;
        
        return -1;
    }
};

File: JS/3550.smallest_index_with_digit_sum_equal_to_index [1].js
/**
 * @param {number[]} nums
 * @return {number}
 */
var smallestIndex = function(nums) {
    function digitSum(num){
        const str = String(num)
        
        let sum = 0
        for(const ch of str)
            sum += ch - '0'

        return sum
    }

    for(let i=0; i<nums.length; i++)
        if(digitSum(nums[i]) === i)
            return i
    
    return -1
};